opentargets_pharmgkb/evidence_generation.py

```diff
@@ -1,44 +1,78 @@
 import json
+import logging
 import multiprocessing
+import os
 from itertools import zip_longest
 
 import pandas as pd
 from cmat.consequence_prediction.common.biomart import query_biomart
 from cmat.consequence_prediction.snp_indel_variants.pipeline import process_variants
 
+from opentargets_pharmgkb.counts import ClinicalAnnotationCounts
 from opentargets_pharmgkb.ontology_apis import get_chebi_iri, get_efo_iri
 from opentargets_pharmgkb.pandas_utils import none_to_nan, explode_column
 from opentargets_pharmgkb.variant_coordinates import get_coordinates_for_clinical_annotation
 
+logging.basicConfig()
+logger = logging.getLogger(__package__)
+logger.setLevel(level=logging.DEBUG)
+
 ID_COL_NAME = 'Clinical Annotation ID'
 
 
-def pipeline(clinical_annot_path, clinical_alleles_path, clinical_evidence_path, drugs_path, created_date, output_path):
+def pipeline(data_dir, created_date, output_path):
+    clinical_annot_path = os.path.join(data_dir, 'clinical_annotations.tsv')
+    clinical_alleles_path = os.path.join(data_dir, 'clinical_ann_alleles.tsv')
+    clinical_evidence_path = os.path.join(data_dir, 'clinical_ann_evidence.tsv')
+    drugs_path = os.path.join(data_dir, 'drugs.tsv')
+    for p in (clinical_annot_path, clinical_alleles_path, clinical_evidence_path, drugs_path):
+        if not os.path.exists(p):
+            logger.error(f'Missing required data file: {p}')
+            raise ValueError(f'Missing required data file: {p}')
+
     clinical_annot_table = read_tsv_to_df(clinical_annot_path)
     clinical_alleles_table = read_tsv_to_df(clinical_alleles_path)
     clinical_evidence_table = read_tsv_to_df(clinical_evidence_path)
     drugs_table = read_tsv_to_df(drugs_path)
 
-    merged_table = pd.merge(clinical_annot_table, clinical_alleles_table, on=ID_COL_NAME, how='left')
     # Restrict to variants with rsIDs
-    rs_only_table = merged_table[merged_table['Variant/Haplotypes'].str.contains('rs')]
+    rs_only_table = clinical_annot_table[clinical_annot_table['Variant/Haplotypes'].str.contains('rs')]
 
-    coordinates_table = get_vcf_coordinates(rs_only_table)
+    # Gather input counts
+    counts = ClinicalAnnotationCounts()
+    counts.clinical_annotations = len(clinical_annot_table)
+    counts.with_rs = len(rs_only_table)
+
+    # Main processing
+    merged_with_alleles_table = pd.merge(rs_only_table, clinical_alleles_table, on=ID_COL_NAME, how='left')
```

Comment:
Should we count after the merge with allele? I'm somehow thinking that this is the source of explosion in number of evidence rather than the explosion by consequence/drug/phenotype.

Reply:
Good point, this is almost certainly the case: not all records have multiples of the other fields but I bet virtually all them have at least 3 alleles. I'll add this.

Reply:
Updated counts in the description.